--- docroot/modules/contrib/addtocal/src/Plugin/Field/FieldFormatter/AddtocalView.php	2024-07-02 10:48:34.997539200 -0500
+++ docroot/modules/contrib/addtocal/src/Plugin/Field/FieldFormatter/AddtocalView.php	2024-07-02 10:24:09.212000000 -0500
@@ -255,11 +255,6 @@
             'aria-label' => $this->t('Add to Yahoo Calendar'),
             'generator' => new Yahoo(),
           ],
-          'web_outlook' => [
-            'title' => $this->t('Outlook.com'),
-            'aria-label' => $this->t('Add to Outlook.com Calendar'),
-            'generator' => new WebOutlook(),
-          ],
           'web_office' => [
             'title' => $this->t('Office.com'),
             'aria-label' => $this->t('Add to Office.com Calendar'),
