
   
diff --git a/js/plugins/link/plugin.js b/js/plugins/link/plugin.js
index 6faa525..b0cbc62 100644
--- a/js/plugins/link/plugin.js
+++ b/js/plugins/link/plugin.js
@@ -73,7 +73,7 @@
 			editor.setKeystroke( CKEDITOR.CTRL + 76 /*L*/, 'link' );
 
 			// (#2478)
-			editor.setKeystroke( CKEDITOR.CTRL + 75 /*K*/, 'link' );
+			//editor.setKeystroke( CKEDITOR.CTRL + 75 /*K*/, 'link' );
 
 			if ( editor.ui.addButton ) {
 				editor.ui.addButton( 'Link', {