diff --git a/src/ContentExporter.php b/src/ContentExporter.php
index 02fdcf37..2aa27c4c 100644
--- a/src/ContentExporter.php
+++ b/src/ContentExporter.php
@@ -18,6 +18,7 @@
 use Drupal\field\FieldConfigInterface;
 use Drupal\layout_builder\Plugin\Block\InlineBlock;
 use Drupal\media\MediaInterface;
+use Drupal\menu_link_content\Entity\MenuLinkContent;
 use Drupal\node\NodeInterface;
 use Drupal\taxonomy\TermInterface;
 use Drupal\user\UserInterface;
@@ -269,7 +270,7 @@ public function exportBaseValues(FieldableEntityInterface $entity): array {
       case 'node':
         if ($entity instanceof NodeInterface) {
           $owner = $entity->getOwner();
-
+          $array_menu_link = $this->generateMenuLink($entity);
           return [
             'title' => $entity->getTitle(),
             'status' => $entity->isPublished(),
@@ -279,6 +280,7 @@ public function exportBaseValues(FieldableEntityInterface $entity): array {
             'url' => $entity->hasField('path') ? $entity->get('path')->alias : NULL,
             'revision_log_message' => $entity->getRevisionLogMessage(),
             'revision_uid' => $entity->getRevisionUserId(),
+            'menu_link_content' => $array_menu_link
           ];
         }
         break;
@@ -351,11 +353,16 @@ public function exportBaseValues(FieldableEntityInterface $entity): array {
   public function exportCustomValues(FieldableEntityInterface $entity, bool $check_translated_fields_only = FALSE): array {
     $fields = $check_translated_fields_only ? $entity->getTranslatableFields() : $entity->getFields();
     $values = [];
-
-    foreach ($fields as $field) {
-      if ($field->getFieldDefinition() instanceof FieldConfigInterface) {
+    if ($entity instanceof \Drupal\smart_date_recur\Entity\SmartDateRule || $entity instanceof \Drupal\smart_date_recur\Entity\SmartDateOverride) {
+      foreach ($fields as $field) {
         $values[$field->getName()] = !$field->isEmpty() ? $this->getFieldValue($field) : NULL;
       }
+    } else {
+      foreach ($fields as $field) {
+        if ($field->getFieldDefinition() instanceof FieldConfigInterface) {
+          $values[$field->getName()] = !$field->isEmpty() ? $this->getFieldValue($field) : NULL;
+        }
+      }
     }
 
     return $values;
@@ -375,6 +382,31 @@ public function getFieldValue(FieldItemListInterface $field) {
       case 'address':
       case 'daterange':
       case 'datetime':
+      case 'smartdate':
+        $date = $field->getValue();
+        $ids = array_column($field->getValue(), 'rrule');
+        if (isset($ids) && sizeof($ids) > 0) {
+          $smart_date_rule_storage = $this->entityTypeManager->getStorage('smart_date_rule');
+          $smart_date_override_storage = $this->entityTypeManager->getStorage('smart_date_override');
+          $smart_date_rules = $smart_date_rule_storage->loadMultiple($ids);
+          $smart_date_overrides = $smart_date_override_storage->loadByProperties(['rrule' => $ids]);
+          $rules_array = [];
+          $overrides_array = [];
+          foreach ($smart_date_rules as $rule) {
+            $rules_array[] = $this->doExportToArray($rule);
+          }
+          foreach ($smart_date_overrides as $override) {
+            $overrides_array[] = $this->doExportToArray($override);
+          }
+          $value = [
+            'dates' => $date,
+            'rrule' => $rules_array,
+            'overrides' => $overrides_array
+          ];
+        } else {
+          $value = $field->getValue();
+        }
+        break;
       case 'email':
       case 'geolocation':
       case 'link':
@@ -389,6 +421,10 @@ public function getFieldValue(FieldItemListInterface $field) {
       case 'text':
       case 'string':
       case 'string_long':
+      case 'uuid':
+        $value = $field->getValue();
+        break;
+
       case 'yearonly':
         $value = $field->getValue();
         break;
@@ -575,4 +611,23 @@ public function getFieldValue(FieldItemListInterface $field) {
     return $value;
   }
 
+  public function generateMenuLink($entity) {
+    $nid = $entity->get('nid')->getValue()[0]['value'];
+    $menu_link_manager = \Drupal::service('plugin.manager.menu.link');
+    $result = $menu_link_manager->loadLinksByRoute('entity.node.canonical', array('node' => $nid));
+
+    $menu_link = array_pop($result);
+    if (isset($menu_link)) {
+      return array(
+        'title' => $menu_link->getTitle(),
+        'description' => $menu_link->getDescription(),
+        'menu_name' => $menu_link->getMenuName(),
+        'weight' => $menu_link->getWeight(),
+        'bundle' => 'menu_link_content',
+      );
+    } else {
+      return null;
+    }
+  }
+
 }
