diff --git a/src/ContentImporter.php b/src/ContentImporter.php
index 655f0d2d..1907450f 100644
--- a/src/ContentImporter.php
+++ b/src/ContentImporter.php
@@ -268,6 +268,36 @@ public function setFieldValue(FieldableEntityInterface $entity, $field_name, $fi
       case 'address':
       case 'daterange':
       case 'datetime':
+      case 'smartdate':
+        if (is_array($field_value)){
+          if (isset($field_value['dates'])){
+            foreach ($field_value['dates'] as $index => $value) {
+              if (isset($value['rrule'])) {
+                $field_value['dates'][$index]['rrule'] = $this->increaseRrule($field_value['dates'][$index]['rrule']);
+              }
+            }
+            $entity->set($field_name, $field_value['dates']);
+          }
+          if (isset($field_value['rrule'])){
+            // Save rrule
+            foreach ($field_value['rrule'] as $index => $value) {
+              $field_value['rrule'][$index]['custom_fields']['rid'][0]['value'] = $this->increaseRrule($field_value['rrule'][$index]['custom_fields']['rid'][0]['value']);
+              $this->doImport($field_value['rrule'][$index]);
+            }
+          }
+          if (isset($field_value['overrides'])) {
+            // Save overrides
+            foreach ($field_value['overrides'] as $index => $value) {
+              // TODO - INCREASE RRULE IS WRONG
+              $field_value['overrides'][$index]['custom_fields']['rrule'][0]['value'] = $this->increaseRrule($field_value['overrides'][$index]['custom_fields']['rrule'][0]['value']);
+              $this->doImport($field_value['overrides'][$index]);
+            }
+          }
+        } else {
+          // Save dates
+          $entity->set($field_name, $field_value);
+        }
+        break;
       case 'email':
       case 'geolocation':
       case 'link':
@@ -644,4 +674,15 @@ protected function isFullEntity(array $entity): bool {
     return isset($entity['uuid']) && isset($entity['entity_type']) && isset($entity['base_fields']) && isset($entity['custom_fields']);
   }

+  /**
+   * Updates RRULE IDs to avoid potential collisions
+   * @param $rrule
+   * @return float|int|void
+   */
+  protected function increaseRrule($rrule) {
+    if (isset($rrule) && is_numeric($rrule)) {
+      return $rrule * 1000;
+    }
+  }
+
 }
