--- draggableviews.module.old	2022-03-24 11:15:48.271017200 -0500
+++ draggableviews.module	2022-03-24 11:25:33.659262500 -0500
@@ -219,30 +219,3 @@
     \Drupal::messenger()->addMessage(t('There was an error while saving the data. Please, try again.'), 'warning');
   }
 }
-
-/**
- * Implements hook_views_query_alter().
- */
-function draggableviews_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
-  // Add additional `join condition` for fixing table join.
-  // Cannot add it in the draggableviews_views_data_alter because we need view
-  // data.
-  if ($query instanceof Sql && isset($query->getTableQueue()['draggableviews_structure'])) {
-    /** @var \Drupal\views\Plugin\views\join\Standard $join */
-    $join = $query->getTableQueue()['draggableviews_structure']['join'];
-    $join->extra = [
-      [
-        'field' => 'view_name',
-        'operator' => '=',
-        'value' => $view->id(),
-        'numeric' => FALSE,
-      ],
-      [
-        'field' => 'view_display',
-        'operator' => '=',
-        'value' => $view->current_display,
-        'numeric' => FALSE,
-      ],
-    ];
-  }
-}
