--- draggableviews.module	2022-03-24 15:14:34.636765200 -0500
+++ draggableviews.module	2022-03-24 15:11:42.370883800 -0500
@@ -219,10 +219,10 @@
     \Drupal::messenger()->addMessage(t('There was an error while saving the data. Please, try again.'), 'warning');
   }
 }
-
 /**
  * Implements hook_views_query_alter().
  */
+
 function draggableviews_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
   // Add additional `join condition` for fixing table join.
   // Cannot add it in the draggableviews_views_data_alter because we need view
@@ -234,13 +234,15 @@
       [
         'field' => 'view_name',
         'operator' => '=',
-        'value' => $view->id(),
+        //'value' => $view->id(),
+        'value' => 'staff_order',
         'numeric' => FALSE,
       ],
       [
         'field' => 'view_display',
         'operator' => '=',
-        'value' => $view->current_display,
+        //'value' => $view->current_display,
+        'value' => 'page_1',
         'numeric' => FALSE,
       ],
     ];
