diff --git a/src/Plugin/migrate/destination/DraggableViews.php b/src/Plugin/migrate/destination/DraggableViews.php
index 1dfac33..fad440b 100644
--- a/src/Plugin/migrate/destination/DraggableViews.php
+++ b/src/Plugin/migrate/destination/DraggableViews.php
@@ -3,7 +3,7 @@
 namespace Drupal\draggableviews\Plugin\migrate\destination;
 
 use Drupal\Core\Database\Database;
-use Drupal\migrate\Entity\MigrationInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
 use Drupal\migrate\Plugin\migrate\destination\DestinationBase;
 use Drupal\migrate\Row;
 
