--- docroot/modules/contrib/group_content_menu/src/Entity/GroupContentMenu.php	2025-08-26 14:57:52.828924200 -0500
+++ docroot/modules/contrib/group_content_menu/src/Entity/GroupContentMenu.php.new.php	2025-08-26 15:02:31.900895600 -0500
@@ -42,14 +42,14 @@
  *     "uuid" = "uuid"
  *   },
  *   links = {
- *     "add-form" = "/group/{group}/menu/add/{plugin_id}",
- *     "add-page" = "/group/{group}/menu/add",
- *     "add-menu-link" = "/group/{group}/menu/{group_content_menu}/add-link",
- *     "edit-menu-link" = "/group/{group}/menu/{group_content_menu}/link/{menu_link_content}",
- *     "delete-menu-link" = "/group/{group}/menu/{group_content_menu}/link/{menu_link_content}/delete",
- *     "edit-form" = "/group/{group}/menu/{group_content_menu}/edit",
- *     "delete-form" = "/group/{group}/menu/{group_content_menu}/delete",
- *     "collection" = "/group/{group}/menus"
+ *     "add-form" = "/group/{group}/content-menu/add/{plugin_id}",
+ *     "add-page" = "/group/{group}/content-menu/add",
+ *     "add-menu-link" = "/group/{group}/content-menu/{group_content_menu}/add-link",
+ *     "edit-menu-link" = "/group/{group}/content-menu/{group_content_menu}/link/{menu_link_content}",
+ *     "delete-menu-link" = "/group/{group}/content-menu/{group_content_menu}/link/{menu_link_content}/delete",
+ *     "edit-form" = "/group/{group}/content-menu/{group_content_menu}/edit",
+ *     "delete-form" = "/group/{group}/content-menu/{group_content_menu}/delete",
+ *     "collection" = "/group/{group}/content-menus"
  *   },
  *   bundle_entity_type = "group_content_menu_type",
  *   field_ui_base_route = "entity.group_content_menu_type.edit_form"
