--- GroupMenuConfigOverrides.php	2022-04-22 14:00:58.788624700 -0500
+++ GroupMenuConfigOverrides_new.php	2022-05-23 11:04:01.706846900 -0500
@@ -210,7 +210,8 @@
     $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid] = [];
     foreach ($group_contents as $group_content) {
       /** @var \Drupal\group\Entity\GroupContentInterface $group_content */
-      if ($group_content && $group_content->getGroup()->hasPermission("update $plugin_id entity", $this->currentUser) && $entity = $group_content->getEntity()) {
+      $group = $group_content->getGroup();
+      if (($group !== NULL) && $group->hasPermission("update $plugin_id entity", $this->currentUser) && $group_content->getEntity()) {
         $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid][] = $entity->id();
       }
     }
