--- src/GroupMenuConfigOverrides.php	2022-05-24 11:12:30.282628100 -0500
+++ src/GroupMenuConfigOverrides.php	2022-05-25 07:58:49.537607900 -0500
@@ -215,8 +215,9 @@
       }
     }
 
-    $this->cache->set($cid, $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid]);
-    return $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid];
+    // $this->cache->set($cid, $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid]);
+    
+	return $this->userGroupMenuIds[$this->currentUser->id()][$group_types_cid];
   }
 
   /**
