diff --git a/js/linkit.autocomplete.js b/js/linkit.autocomplete.js
index 558278dd..f417fa02 100644
--- a/js/linkit.autocomplete.js
+++ b/js/linkit.autocomplete.js
@@ -171,6 +171,17 @@
         $autocomplete.autocomplete(autocomplete.options);
         $autocomplete.autocomplete('widget').addClass('linkit-ui-autocomplete');
 
+        // find and add class to any wrappers that do not have it via manually searching. Works
+        // for any number of autocomplete wrappers instead of just one
+        var $parent = $('#drupal-modal');
+        var $children = $parent.children('.ui-autocomplete');
+        for (let i = 0; i < $children.length; i++) {
+          if (!$children[i].classList.contains('linkit-ui-autocomplete')){
+            $children[i].classList.add('linkit-ui-autocomplete');
+            $children[i].classList.remove('ui-autocomplete');
+          }
+        }
+
         $autocomplete.click(function () {
           $autocomplete.autocomplete('search', $autocomplete.val());
         });